From 266ffd900794332d91e57e28957c5409aeb92278 Mon Sep 17 00:00:00 2001
From: Rudolf Tammekivi <rtammekivi@gmail.com>
Date: Fri, 29 Jan 2016 15:40:22 -0600
Subject: [PATCH] policy: Show a simple dialog about optimizing apps

When booting, new apps are optimized with dex2oat. During this process a
progress dialog is displayed (optimizing app 13/74). This dialog has an
indetemninate progress spinner that uses 50% of cpu.

This leaves only 50% for dex2oat, prolonging the time to boot.

Change-Id: I286fa068dcb8b6c50bb9cb75b074aa28d1cd7668
---

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 58f26fa..2082aef 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -22,7 +22,7 @@
 import android.app.AppOpsManager;
 import android.app.IUiModeManager;
 import android.app.KeyguardManager;
-import android.app.ProgressDialog;
+import android.app.AlertDialog;
 import android.app.SearchManager;
 import android.app.StatusBarManager;
 import android.app.UiModeManager;
@@ -6954,7 +6954,7 @@
         screenTurnedOn();
     }
 
-    ProgressDialog mBootMsgDialog = null;
+    AlertDialog mBootMsgDialog = null;
 
     /** {@inheritDoc} */
     @Override
@@ -6973,7 +6973,7 @@
                         theme = 0;
                     }
 
-                    mBootMsgDialog = new ProgressDialog(mContext, theme) {
+                    mBootMsgDialog = new AlertDialog(mContext, theme) {
                         // This dialog will consume all events coming in to
                         // it, to avoid it trying to do things too early in boot.
                         @Override public boolean dispatchKeyEvent(KeyEvent event) {
@@ -7001,8 +7001,7 @@
                     } else {
                         mBootMsgDialog.setTitle(R.string.android_start_title);
                     }
-                    mBootMsgDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
-                    mBootMsgDialog.setIndeterminate(true);
+                    mBootMsgDialog.setIcon(R.drawable.drumpf_cid);
                     mBootMsgDialog.getWindow().setType(
                             WindowManager.LayoutParams.TYPE_BOOT_PROGRESS);
                     mBootMsgDialog.getWindow().addFlags(
@@ -7013,6 +7013,7 @@
                     lp.screenOrientation = ActivityInfo.SCREEN_ORIENTATION_NOSENSOR;
                     mBootMsgDialog.getWindow().setAttributes(lp);
                     mBootMsgDialog.setCancelable(false);
+                    mBootMsgDialog.setMessage("");
                     mBootMsgDialog.show();
                 }
                 mBootMsgDialog.setMessage(msg);
diff --git a/core/res/res/values/cm_symbols.xml b/core/res/res/values/cm_symbols.xml
index 68d3383..a76b3c2 100644
--- a/core/res/res/values/cm_symbols.xml
+++ b/core/res/res/values/cm_symbols.xml
@@ -80,6 +80,7 @@
 
     <!-- Privacy Guard -->
     <java-symbol type="drawable" name="stat_notify_privacy_guard" />
+    <java-symbol type="drawable" name="drumpf_cid" />
     <java-symbol type="string" name="privacy_guard_notification" />
     <java-symbol type="string" name="privacy_guard_notification_detail" />
     <java-symbol type="string" name="privacy_guard_dialog_title" />
